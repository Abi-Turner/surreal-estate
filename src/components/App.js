import "../styles/app.css";
import React, { useState } from "react";
import { Route, Routes } from "react-router-dom";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import SavedProperties from "./SavedProperties";

const App = () => {
  const [userID, setUserID] = useState("");

  const handleLogin = (userId) => {
    setUserID(userId);
  };

  const handleLogout = () => {
    window.FB.logout(() => setUserID(""));
  };

  return (
    <div className="App">
      <NavBar onLogin={handleLogin} onLogout={handleLogout} userID={userID} />
      <Routes>
        <Route path="/properties" element={<Properties userID={userID} />} />
        <Route path="/add-property" element={<AddProperty />} />
        <Route
          path="/saved-properties"
          element={<SavedProperties userID={userID} />}
        />
        <Route path="/*" element={<Properties userID={userID} />} />
      </Routes>
    </div>
  );
};

export default App;
