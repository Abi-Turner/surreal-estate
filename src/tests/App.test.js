import { render } from "@testing-library/react";
import React from "react";
import App from "../components/App";
import { BrowserRouter } from "react-router-dom";

// test("renders app correctly", () => {
//   render(<App />);
//   const linkElement = screen.getByText(/Surreal Estate/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe("App", () => {
  const { asFragment } = render(
    <BrowserRouter basename="/">
      <App />
    </BrowserRouter>
  );

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
