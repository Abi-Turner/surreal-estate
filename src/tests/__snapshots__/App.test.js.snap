// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="nav-bar"
    >
      <img
        alt="logo-icon"
        src="logo.png"
      />
      Surreal Estate
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links-item"
        >
          <a
            class="item"
            href="/properties"
          >
            View Properties
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            class="item"
            href="/add-property"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
